pkg/publisher/publisher.go

```diff
@@ -0,0 +1,390 @@
+package publisher
```

Comment:
This package used to be call `dirpublisher`, so it contains lots of pre-existing code. Not much to worry about in here.